File: processorjavalib/src/main/java/com/masadora/processorjavalib/annotation/CollectionElement.java
package com.masadora.processorjavalib.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * PACK com.masadora.processorjavalib.annotation
 * CREATE BY Shay
 * DATE BY 2022/12/9 14:27 星期五
 * <p>
 * DESCRIBE
 * <p>
 */
// TODO:2022/12/9 
@Retention(RetentionPolicy.SOURCE)
@Target(ElementType.TYPE)
public @interface CollectionElement {
     String collectionName();
}

File: processorlib/src/main/java/com/shay/processorlib/annotation/CollectionElement.kt



/**
 * PACK com.masadoraandroid.site.annotation
 * CREATE BY Shay
 * DATE BY 2022/11/30 18:45 星期三
 * <p>
 * DESCRIBE
 * <p>
 */
// TODO:2022/11/30
@Retention(AnnotationRetention.SOURCE)
@Target(AnnotationTarget.CLASS)
@MustBeDocumented
annotation class CollectionElement(val collectionName: String = "Default")
/*
@Retention(AnnotationRetention.SOURCE)
@Target(AnnotationTarget.CLASS)
annotation class CollectionType<T : Any?>(val collectionType:Class<in T> = Object::class.java)*/
